WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadNotifier.java

```diff
@@ -241,25 +346,100 @@ private synchronized void doNotify(long id, Notification notification) {
     }
 
     void setTotalMediaItems(PostModel post, int totalMediaItems) {
-        NotificationData notificationData = sPostIdToNotificationData.get(post.getId());
-        if (totalMediaItems <= 0) {
-            totalMediaItems = 1;
-        }
+        sNotificationData.totalMediaItems+=totalMediaItems;
 
-        notificationData.totalMediaItems = totalMediaItems;
-        notificationData.itemProgressSize = 100.0f / notificationData.totalMediaItems;
+        // TODO MEDIA REIMPLEMENT OR DELETE THIS
+//        NotificationData notificationData = sPostIdToNotificationData.get(post.getId());
+//        notificationData.totalMediaItems = totalMediaItems;
+//        notificationData.itemProgressSize = 100.0f / notificationData.totalMediaItems;
     }
 
     void setCurrentMediaItem(PostModel post, int currentItem) {
-        NotificationData notificationData = sPostIdToNotificationData.get(post.getId());
-        notificationData.currentMediaItem = currentItem;
-
-        mNotificationBuilder.setContentText(String.format(mContext.getString(R.string.uploading_total),
-                currentItem, notificationData.totalMediaItems));
+        // TODO MEDIA REIMPLEMENT OR DELETE THIS
     }
 
     private String buildNotificationTitleForPost(PostModel post) {
         String postTitle = TextUtils.isEmpty(post.getTitle()) ? mContext.getString(R.string.untitled) : post.getTitle();
         return String.format(mContext.getString(R.string.uploading_post), postTitle);
     }
+
+    private String buildNotificationTitleForMedia() {
+        return mContext.getString(R.string.uploading_post_media);
+    }
+
+    private String buildNotificationTitleForMixedContent() {
+        return mContext.getString(R.string.uploading_title);
+    }
+
+    private String buildNotificationSubtitleForPost(PostModel post){
+        String uploadingMessage = String.format(
+                mContext.getString(R.string.uploading_subtitle_posts_only),
+                getCurrentPostItem() + 1,
+                sNotificationData.totalPostItems,
+                post.isPage() ? mContext.getString(R.string.page).toLowerCase()
+                        : mContext.getString(R.string.post).toLowerCase()
+        );
+        return uploadingMessage;
+    }
+
+    private String buildNotificationSubtitleForPosts(){
+        String pagesAndOrPosts = getPagesAndOrPostsString();
+        String uploadingMessage = String.format(
+                mContext.getString(R.string.uploading_subtitle_posts_only),
+                getCurrentPostItem() + 1,
+                sNotificationData.totalPostItems,
+                pagesAndOrPosts
+        );
+        return uploadingMessage;
+    }
+
+    private String getPagesAndOrPostsString() {
+        String pagesAndOrPosts = "";
```

Comment:
Minor nit: it's not necessary to initialize the string to `""` here.

Reply:
addressed in faeb5a8